### PR TITLE
Implement syntax highlighter

### LINKED ISSUE #9: [Project - Onboarding]: Implement Code Syntax highlighter (without styling)
# Implement React Component: CodeHighlighter

## Objective:
The objective of this issue is to create a new React component named `CodeSyntaxHighlighter`. It is used to highlight code within a box.

#### 1. Properties:

```tsx
interface Code {
  id: string
  code: string
}
```

| Property  | Type | Description | Required |
| --------- | ---- | ----------- | -------- |
| codes  | Array<Code> | Code snippets | Yes |
| activeIDs | string[] | The Code snippet ids which should be highlighted | Yes |

## Alternative Objective:
Since the code which should be highlighted is known, we can also "fake" the highlighting. We don't have to introduce a complex library or we don't have to change/adjust the existing CodeHighlighter of the Monorepo to fit its usage in this particular case.

## Behaviors:

- The Code highlighter should only highlight the active code snippets.
- All the code is rendered in a container.
- The code parts should be placed separately to the DOM (with ID's) to be able to highlight and select them independently. 
- The Code Box is scrollable if the maximum height is reached.

### 3. Design (DRAFT):
image

## Acceptance Criteria:

- The component is implemented according to the requirements described above, or an existing component is taken that fulfills the requirements.
- The component passes all defined test cases and has Stories displaying its states
- The component linked to a figma component in Storybook - speak to designers!
- You should not implement any complex CSS styles! The design is still a work in progress. The component will be finalized design-wise in a separate ticket. Ensure that the `properties` and the `behaviors` are implemented accordingly.
- Discuss with the design-team which highlight theme should be used. Let's avoid a custom-made one.

### PR DESCRIPTION
**Pull request by valentinpalkovic**
Closes https://github.com/storybookjs/addon-onboarding/issues/9

## What I did

I have implemented a generic syntax highlighter, which can be used in the "How to write your first Story" flow.

<details>
  <summary>📦 Published PR as canary version: <code>0.0.13--canary.20.b5f373d.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-onboarding@0.0.13--canary.20.b5f373d.0
  # or 
  yarn add @storybook/addon-onboarding@0.0.13--canary.20.b5f373d.0
  ```
</details>

### CODE REVIEW

File: src/components/SyntaxHighlighter/Snippet/Snippet.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                </StorybookSyntaxHighlighter>
+              )}
+
+              {toggle && !open && (
+                <StorybookSyntaxHighlighter
+                  key={i}
+                  language="javascript"
+                  customStyle={{ fontSize: "0.8rem" }}
+                >
+                  {`  ...`}
```
Comment (yannbf) [inline src/components/SyntaxHighlighter/Snippet/Snippet.tsx]:
```suggestion
                  {`  // ...`}
```

File: src/components/SyntaxHighlighter/SyntaxHighlighter.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+import { Button } from './Button';`,
+    },
+  ],
+  [
+    {
+      content: `const meta: Meta<typeof Button> = {
+  title: 'Example/Button',
+  component: Button,
+  ...
```
Comment (yannbf) [inline src/components/SyntaxHighlighter/SyntaxHighlighter.stories.tsx]:
```suggestion
  // ...
```

File: src/components/SyntaxHighlighter/SyntaxHighlighter.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+      toggle: true,
+    },
+    { content: `};` },
+  ],
+  [
+    {
+      content: `// Copy the code below
+
+  export const Warning: Story = {
+    args: {
```
Comment (yannbf) [inline src/components/SyntaxHighlighter/SyntaxHighlighter.stories.tsx:60]: There's a small formatting issue here
image

**Comment by yannbf** [comment]
This looks pretty awesome <3

I wonder if the component should be responsive? I think it will have some trouble when in the modal:
image

**Comment by valentinpalkovic** [comment]
> This looks pretty awesome <3
> 
> I wonder if the component should be responsive? I think it will have some trouble when in the modal: image

We don't want to line wrap the code. Instead, the code is scrollable horizontally, if this case happens.

**Pull request merged**